sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithStateSuite.scala

```diff
@@ -133,6 +133,98 @@ class RunningCountStatefulProcessorWithAddRemoveProcTimeTimer
   }
 }
 
+// Class to verify stateful processor usage with adding event time timers
+class RunningCountStatefulProcessorWithEventTimeTimer
+  extends StatefulProcessor[String, (String, java.sql.Timestamp), (String, String)] {
+
+  @transient var _countState: ValueState[Long] = _
+  @transient var _processorHandle: StatefulProcessorHandle = _
+
+  override def init(
+       handle: StatefulProcessorHandle,
+       outputMode: OutputMode): Unit = {
+    _processorHandle = handle
+    assert(handle.getQueryInfo().getBatchId >= 0)
+    assert(handle.getQueryInfo().getOperatorId == 0)
+    assert(handle.getQueryInfo().getPartitionId >= 0 && handle.getQueryInfo().getPartitionId < 5)
```

Comment:
Can you merge back once from my branch ? I think I added a utils object for storing numPartitions.